ctapipe/reco/shower_processor.py

```diff
@@ -0,0 +1,197 @@
+"""
+High level processing of showers.
+
+This processor will be able to process a shower/event in 3 steps:
+- shower geometry
+- estimation of energy (optional, currently unavailable)
+- estimation of classification (optional, currently unavailable)
+
+"""
+from traitlets import Bool
+
+from astropy.coordinates import SkyCoord, AltAz
+
+from ctapipe.core import Component, QualityQuery
+from ctapipe.core.traits import List
+from ctapipe.containers import ArrayEventContainer, ReconstructedShowerContainer
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.reco import HillasReconstructor
+
+
+__all__ = ["ShowerQualityQuery", "ShowerProcessor"]
+
+
+DEFAULT_SHOWER_PARAMETERS = ReconstructedShowerContainer()
+
+
+class ShowerQualityQuery(QualityQuery):
+    """Configuring shower-wise data checks."""
+
+    quality_criteria = List(
+        default_value=[
+            ("> 50 phe", "lambda p: p.hillas.intensity > 50"),
+            ("Positive width", "lambda p: p.hillas.width.value > 0"),
+            ("> 3 pixels", "lambda p: p.morphology.num_pixels > 3"),
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+
+class ShowerProcessor(Component):
+    """
+    Needs DL1_PARAMETERS as input.
+    Should be run after ImageProcessor to produce all DL1b information.
+
+    For the moment it only supports the reconstruction of the shower geometry
+    using ctapipe.reco.HillasReconstructor.
+
+    It is planned to support also energy reconstruction and particle type
```

Comment:
Then don't add them here please. We made bad experience with stubs. Just don't implement anything that is not yet implemented please.

Reply:
this (and the `raise not ImplementedError`s ) was suggested by @kosack but I have not strong preference for now

Reply:
We made pretty bad experience with this kind of premature API design in the past, I'd say. 

It's not usable now and it might elad to restricting the actual imlpementation when it's done unnecessarily. It's just dead code. And it confuses users.

The methods being there and documented might on first glance lead to users thinking it's already implemented, only to be then badly surprised by `NotImplemented`.

I am really not a fan of having `raise NotImplementedError` functions / methods.

Reply:
You can remove them - in the end the only function the user should call is  the __call__ method anyhow, so if new features are added, it doesn't break anything.  It is important I think to remember that this is intended to also do energy and particle type reco,  so mentioning it in the doc is I think fine. (in case somebody tries to implement it outside).

Reply:
done